port-chooser/src/main/java/org/terracotta/utilities/test/net/PortManager.java

```diff
@@ -433,6 +436,26 @@ private boolean refusesConnect(int port) {
     return isFree;
   }
 
+  /**
+   * Checks that the port being released is actually free and emits a detailed log message
+   * if not.
+   * @param port the port to check
+   */
+  private void diagnosticReleaseCheck(int port) {
+    try {
+      List<NetStat.BusyPort> collisions = NetStat.info().stream()
+          .filter(p -> p.localEndpoint().getPort() == port)
+          .collect(toList());
+      if (!collisions.isEmpty()) {
+        LOGGER.error("Port {} being released to PortManager is in use by the following:{}",
+            port, collisions.stream().map(NetStat.BusyPort::toString)
+                .collect(joining("\n    ", "\n    ", "")));
```

Comment:
I think I'd prefer if we:

-  moved the leading newline to the message.
-  mapped the objects to space prefixed strings
-  joined with `joining('\n')